android/anc/src/main/java/org/smartregister/fhircore/anc/data/patient/PatientRepository.kt

```diff
@@ -380,29 +381,46 @@ constructor(
     questionnaireResponse: QuestionnaireResponse,
     patientId: String,
     encounterID: String,
-    height: Double,
     weight: Double,
-    computedBmi: Double
+    height: Double,
+    computedBmi: Double,
+    isUnitModeMetric: Boolean
   ): Boolean {
-    resourceMapperExtended.saveParsedResource(questionnaireResponse, questionnaire, patientId, null)
-    return recordBmi(patientId, encounterID, height, weight, computedBmi)
+    // resourceMapperExtended.saveParsedResource(questionnaireResponse,
+    // questionnaire, patientId, null)
+    return recordBmi(
+      patientId = patientId,
+      formEncounterId = encounterID,
+      weight = weight,
+      height = height,
+      computedBMI = computedBmi,
+      isUnitModeMetric = isUnitModeMetric
+    )
   }
 
   private suspend fun recordBmi(
     patientId: String,
     formEncounterId: String,
-    height: Double? = null,
     weight: Double? = null,
-    computedBMI: Double? = null
+    height: Double? = null,
+    computedBMI: Double? = null,
+    isUnitModeMetric: Boolean
   ): Boolean {
-    val bmiEncounterData =
-      buildBmiConfigData(
-        patientId = patientId,
-        recordId = formEncounterId,
-        height = height,
-        weight = weight,
-        computedBmi = computedBMI
-      )
+    var weightUnit = "lb"
+    var heightUnit = "in"
+    var weightUnitCode = "[lb_av]"
+    var heightUnitCode = "[in_i]"
+    // Todo: confirm if bmi unit can be displayed in lb/in2 or only in kg/m2
+    var bmiUnit = "kg/m2"
+    if (isUnitModeMetric) {
+      weightUnit = "kg"
+      heightUnit = "cm"
+      weightUnitCode = "kg"
+      heightUnitCode = "cm"
+      bmiUnit = "kg/m2"
```

Comment:
magic string could be replaced with constants

Reply:
updated